CMakeLists.txt

```diff
@@ -199,13 +199,23 @@ SET(ENABLE_RTMP TRUE CACHE BOOL "Enable librtmp and dependent functionality?")
 SET(ENABLE_LLVM FALSE CACHE BOOL "Enable support for llvm based jit execution (currently broken)")
 SET(ENABLE_PROFILING FALSE CACHE BOOL "Enable profiling support? (Causes performance issues)")
 SET(ENABLE_MEMORY_USAGE_PROFILING FALSE CACHE BOOL "Enable profiling of memory usage? (Causes performance issues)")
-SET(PLUGIN_DIRECTORY "${LIBDIR}/mozilla/plugins" CACHE STRING "Directory to install Firefox plugin to")
 SET(PPAPI_PLUGIN_DIRECTORY "${LIBDIR}/PepperFlash" CACHE STRING "Directory to install PPAPI plugin to")
 SET(MANUAL_DIRECTORY "share/man" CACHE STRING "Directory to install manual to (UNIX only)")
 SET(ENABLE_SSE2 TRUE CACHE BOOL "Enable use of SSE2 asm instructions (x86/x86_64 only)")
 
-IF(ENABLE_DEBIAN_ALTERNATIVES OR WIN32)
+# Try to detect where we should install the plugin to, as Firefox version 62+ uses a different directory
+# Unset it first in case we've changed browsers or something between builds
+UNSET(PLUGIN_DIRECTORY CACHE)
+IF(UNIX)
+  EXECUTE_PROCESS(COMMAND firefox --version COMMAND grep -Eo "[0-9]+" COMMAND head -n 1 OUTPUT_VARIABLE FIREFOX_MAJOR_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+  MESSAGE(STATUS "Firefox version ${FIREFOX_MAJOR_VERSION} detected")
+ENDIF()
+IF(FIREFOX_MAJOR_VERSION GREATER 62)
+  SET(PLUGIN_DIRECTORY "$ENV{HOME}/.mozilla/plugins" CACHE STRING "Directory to install Firefox plugin to")
+ELSEIF(ENABLE_DEBIAN_ALTERNATIVES OR WIN33)
```

Comment:
This should be WIN32, I suppose.

Reply:
Not sure how that got in - yes it should be, will change, thanks.